models/enterprise_sonic/bgp_af/replaced_example_01.txt

```diff
@@ -0,0 +1,177 @@
+#  advertise ipv4 unicast route-map bb
+#  rd 1.1.1.1:11
+#  route-target import 12:12
+#  route-target export 13:13
+#  dup-addr-detection
+#  advertise-pip ip 3.3.3.3 peer-ip 4.4.4.4
+#  !
+#  vni 1
+#   advertise-default-gw
+#   advertise-svi-ip
+#   rd 5.5.5.5:55
+#   route-target import 88:88
+#   route-target export 77:77
+
+- name: Replace device configuration of address families of specified BGP AS with provided configuration.
+  dellemc.enterprise_sonic.sonic_bgp_af:
+    config:
+      - bgp_as: 51
+        address_family:
+          afis:
+            - afi: l2vpn
+              safi: evpn
+              advertise_pip: True
+              advertise_pip_ip: "3.3.3.3"
+              advertise_pip_peer_ip: "4.4.4.4"
+              advertise_svi_ip: True
+              advertise_all_vni: False
+              advertise_default_gw: False
+              route_advertise_list:
+                - advertise_afi: ipv4
+                  route_map: bb
+              rd: "1.1.1.1:11"
+              rt_in:
+                - "22:22"
+              rt_out:
+                - "13:13"
+              vnis:
+                - vni_number: 5
+                  advertise_default_gw: True
+                  advertise_svi_ip: True
+                  rd: "10.10.10.10:55"
+                  rt_in:
+                    - "88:88"
+                  rt_out:
+                    - "77:77"
+            - afi: ipv4
+              safi: unicast
+              network:
+                - 2.2.2.2/16
+                - 192.168.10.1/32
+              dampening: True
+              redistribute:
+                - protocol: connected
+                - protocol: ospf
+                  metric: 30
+              aggregate-address-config:
+                - prefix: '5.5.5.5/5'
+                  as_set: True
+      - bgp_as: 51
+        vrf_name: VrfReg2
+        address_family:
+          afis:
+            - afi: ipv4
+              safi: unicast
+              import:
+                vrf:
+                  vrf_list:
+                    - VrfReg1
+                  route_map: rmap-reg1
+    state: replaced
+
+# After state:
+# ------------
+#
+#do show running-configuration bgp
+#!
+#router bgp 51 vrf VrfReg1
+# log-neighbor-changes
+# timers 60 180
+# !
+# address-family ipv4 unicast
+#  maximum-paths 1
+#  maximum-paths ibgp 1
+#  network 3.3.3.3/16
+#  dampening
+#!
+#router bgp 51 vrf VrfReg2
+# log-neighbor-changes
+# timers 60 180
+# !
+# address-family ipv4 unicast
+#  maximum-paths 1
+#  maximum-paths ibgp 1
+#  import vrf route-map rmap-reg1
+#  import vrf VrfReg1
+#!
+#router bgp 51
+# router-id 111.2.2.41
+# log-neighbor-changes
+# timers 60 180
+# !
+# address-family ipv4 unicast
+#  redistribute connected
+#  redistribute ospf metric 30
+#  maximum-paths 1
+#  maximum-paths ibgp 1
+#  network 2.2.2.2/16
+#  network 192.168.10.1/32
+#  aggregate-address 5.5.5.5/5 as-set
+#  dampening
+# !
+# address-family l2vpn evpn
+#  advertise-all-vni
```

Comment:
advertise-all-vni was set to False by the example playbook, so it shouldn't be shown here.

Reply:
Resolved.